packages/supabase/lib/src/constants.dart

```diff
@@ -1,7 +1,18 @@
 import 'package:supabase/src/version.dart';
+import 'dart:io' show Platform;
 
 class Constants {
-  static const Map<String, String> defaultHeaders = {
+  static String get platform {
+    return Platform.operatingSystem;
```

Comment:
I know this is still draft, but still want to mention that this won't work on web.
You will either need a conditional import, or probably better use a constant to guard web with a definition like the `kIsWeb` in flutter:
```dart
const bool kIsWeb = bool.fromEnvironment('dart.library.js_util');
```

Reply:
Thanks, @Vinzent03 

Do we have any test for asserting that it doesn't break in Web? Didn't see any fail on CI.

Reply:
Sadly not yet, but hopefully #1140 solves this

Reply:
@Vinzent03 I added the guard as you suggested, we could send data in case of web, if that is easy to do, not a requirement since our reports also fetches from the `user-agent`, and `user-agent` in web is better defined then mobile.

Reply:
Yeah, I think using the user-agent is the better and correct way for web.